### PR TITLE
[MultiDB]:swsscommon replace old API with new APIs including tests

### PR DESCRIPTION
**Pull request by dzhangalibaba**
* replacing swsscommon old APIs with new APIs, including logger, WarmStart and Unit tests
    * redisign WarmStart::init() signature, reuse new DBConnector API flag to choose SOCKET or TCP connection.
    * unit test adding main() entry function to run multiDB tests first which will init all the db config data once, all other tests will reuse these data.
* update db config file to the latest version which include the separator.
* VS tests all passed.
* swsscommon UT all passed. 

Signed-off-by: Dong Zhang d.zhang@alibaba-inc.com

### CODE REVIEW

**Comment by dzhangalibaba** [comment]
@qiluo-msft swsscommon replacement is ready to go.

File: tests/main.cpp [hunk 1]
```diff
@@ -0,0 +1,12 @@
+#include "gtest/gtest.h"
+
+int main(int argc, char* argv[])
+{
+  testing::InitGoogleTest(&argc, argv);
```
Comment (qiluo-msft) [inline tests/main.cpp]: >   [](start = 0, length = 2)

Please follow code format in other files

File: common/warm_restart.h [hunk 2]
```diff
@@ -43,8 +43,7 @@ class WarmStart
     static void initialize(const std::string &app_name,
```
Comment (qiluo-msft) [inline common/warm_restart.h]: >initialize [](start = 16, length = 10)

Will it break other repos? Do we need to block this PR on other PR?

Comment (dzhangalibaba) [reply common/warm_restart.h]: no. Besides in swsscommon testing cases, only swss OA use this WarmStart::init(), swss already replaced with new APIs.

Comment (qiluo-msft) [reply common/warm_restart.h]: I see. swss is using this with default parameter values.

---
In reply to: [357323615](https://github.com/Azure/sonic-swss-common/pull/324#discussion_r357323615) [](ancestors = 357323615)

File: tests/main.cpp (hunk 1 above)
Comment (dzhangalibaba) [reply tests/main.cpp]: DONE

File: tests/main.cpp [hunk 3]
```diff
@@ -0,0 +1,12 @@
+#include "gtest/gtest.h"
+
+int main(int argc, char* argv[])
+{
+    testing::InitGoogleTest(&argc, argv);
+    //run multiDB test first, which init the db config
```
Comment (qiluo-msft) [inline tests/main.cpp]: >//run multiDB test first, which init the db config [](start = 4, length = 50)

I could understand your intention to order the test cases. I suspect this only works if you run all the test cases, but not if using `--gtest_filter`.

I suggest explore (Global Set-Up and Tear-Down)[[https://github.com/google/googletest/blob/master/googletest/docs/advanced.md#global-set-up-and-tear-down](https://github.com/google/googletest/blob/master/googletest/docs/advanced.md#global-set-up-and-tear-down)], which is design for just this use case.

Comment (dzhangalibaba) [reply tests/main.cpp]: can we have a quick chat, didn't fully understand the question

Comment (dzhangalibaba) [reply tests/main.cpp]: > > //run multiDB test first, which init the db config [](start = 4, length = 50)
> 
> I could understand your intention to order the test cases. I suspect this only works if you run all the test cases, but not if using `--gtest_filter`.
> 
> I suggest explore [Global Set-Up and Tear-Down](https://github.com/google/googletest/blob/master/googletest/docs/advanced.md#global-set-up-and-tear-down), which is design for just this use case.

DONE. tested as below: now multi_db_test and other tests are independent.

```
dzhang@b5f47bc8ad94:/sonic/src/sonic-swss-common$ ./tests/tests --gtest_filter=DBConnector.multi_db_test
Note: Google Test filter = DBConnector.multi_db_test
[==========] Running 1 test from 1 test case.
[----------] Global test environment set-up.
Default : isInit = 0
INIT: loading nonexisting db config file
INIT: load local db config file, isInit = 1
[----------] 1 test from DBConnector
[ RUN      ] DBConnector.multi_db_test
INIT: loading local config file again
GET: invalid dbname input for getDbInst()
GET: invalid dbname input for getDbId()
GET: invalid dbname input for getDbSock()
GET: invalid dbname input for getDbHostname()
GET: invalid dbname input for getDbPort()
testing APPL_DB
redis#0#/var/run/redis/redis.sock#127.0.0.1#6379
testing ASIC_DB
redis#1#/var/run/redis/redis.sock#127.0.0.1#6379
testing CONFIG_DB
redis#4#/var/run/redis/redis.sock#127.0.0.1#6379
testing COUNTERS_DB
redis#2#/var/run/redis/redis.sock#127.0.0.1#6379
testing FLEX_COUNTER_DB
redis#5#/var/run/redis/redis.sock#127.0.0.1#6379
testing LOGLEVEL_DB
redis#3#/var/run/redis/redis.sock#127.0.0.1#6379
testing PFC_WD_DB
redis#5#/var/run/redis/redis.sock#127.0.0.1#6379
testing SNMP_OVERLAY_DB
redis#7#/var/run/redis/redis.sock#127.0.0.1#6379
testing STATE_DB
redis#6#/var/run/redis/redis.sock#127.0.0.1#6379
testing TEST_DB
redis#15#/var/run/redis/redis.sock#127.0.0.1#6379
[       OK ] DBConnector.multi_db_test (1 ms)
[----------] 1 test from DBConnector (1 ms total)

[----------] Global test environment tear-down
[==========] 1 test from 1 test case ran. (1 ms total)
[  PASSED  ] 1 test.
dzhang@b5f47bc8ad94:/sonic/src/sonic-swss-common$ 

```



```
dzhang@b5f47bc8ad94:/sonic/src/sonic-swss-common$ ./tests/tests --gtest_filter=DBConnector.multitable
Note: Google Test filter = DBConnector.multitable
[==========] Running 1 test from 1 test case.
[----------] Global test environment set-up.
Default : isInit = 0
INIT: loading nonexisting db config file
INIT: load local db config file, isInit = 1
[----------] 1 test from DBConnector
[ RUN      ] DBConnector.multitable
Starting 128 producers and consumers on redis, using single thread for consumers and thread per producer
++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
Done.
[       OK ] DBConnector.multitable (7572 ms)
[----------] 1 test from DBConnector (7572 ms total)

[----------] Global test environment tear-down
[==========] 1 test from 1 test case ran. (7572 ms total)
[  PASSED  ] 1 test.

```

**Pull request merged**